fictadvisor-api/src/v2/api/services/UserService.ts

```diff
@@ -359,6 +364,34 @@ export class UserService {
 
   }
 
+  async linkGoogle(userId: string, idToken: string) : Promise<void> {
+    if (!(await this.googleAuthService.isIdTokenValid(idToken))) {
+      throw new InvalidGoogleTokenException();
+    }
+
+    const googleUser = this.googleAuthService.getUserPayload(idToken);
+
+    if (!googleUser.email_verified) {
+      throw new GoogleEmailNotVerifiedException();
+    }
+
+    const sameGoogleAccount = await this.userRepository.find({ googleId: googleUser.sub });
+    if (sameGoogleAccount) {
+      throw new DuplicateGoogleIdException();
+    }
+
+    const user = await this.userRepository.findById(userId);
+    if (user.googleId) {
```

Comment:
Not sure that is necessary as we do not have log out and user can link google account once